src/gl/canvas2dRe.re

```diff
@@ -0,0 +1,76 @@
+type t;
+
+/* Canvas API, following https://simon.html5.org/dump/html5-canvas-cheat-sheet.html */
+
+external save : unit = "" [@@bs.send.pipe: t];
+external restore : unit = "" [@@bs.send.pipe: t];
+
+/* Transformation */
+external scale : float => float => unit = "" [@@bs.send.pipe: t];
+external rotate : float => unit = "" [@@bs.send.pipe: t];
+external translate : float => float => unit = "" [@@bs.send.pipe: t];
+external transform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+external setTransform : m11::float => m12::float => m21::float => m22::float => dx::float => dy::float => unit = "" [@@bs.send.pipe: t];
+
+/* Compositing */
+external globalAlpha : t => float => unit = "" [@@bs.set];
+external globalCompositeOperation : t => string => unit = "" [@@bs.set];
```

Comment:
@glennsl I'm trying to figure out how to type the enums with bucklescript but keep getting stuck. The bucklescript manual uses variant constructors starting with a backtick, which Reason seems to complain about. Here's what I have as an example:

```
type compositeT =
  | SourceOver [@bs.as "source-over"]
  | SourceIn [@bs.as "source-in"]
  | SourceOut [@bs.as "source-out"]
  | SourceAtop [@bs.as "source-atop"]
  | DestinationOver [@bs.as "destination-over"]
  | DestinationIn [@bs.as "destination-in"]
  | DestinationOut [@bs.as "destination-out"]
  | DestinationAtop [@bs.as "destination-atop"]
  | Lighter [@bs.as "lighter"]
  | Copy [@bs.as "copy"]
  | Xor [@bs.as "xor"];

/* and then using it: */
```
external globalCompositeOperation : t => compositeT [@bs.string] => unit = "" [@@bs.set];
```

I'm getting `Error: Not a valid  type for [@bs.string]`. Any pointers?